include/dlaf/matrix/panel.h

```diff
@@ -93,10 +93,12 @@ struct Panel<axis, const T, D> {
 #if defined DLAF_ASSERT_MODERATE_ENABLE
     {
       const auto panel_tile_size = tileSize(index);
-      new_tile_fut.then(pika::launch::sync, pika::unwrapping([panel_tile_size](const auto& tile) {
-                          DLAF_ASSERT_MODERATE(panel_tile_size == tile.size(), panel_tile_size,
-                                               tile.size());
-                        }));
+      auto assert_tile_size = pika::unwrapping([panel_tile_size](ConstTileType const& tile) {
+        DLAF_ASSERT_MODERATE(panel_tile_size == tile.size(), panel_tile_size, tile.size());
+      });
+      pika::execution::experimental::keep_future(new_tile_fut) |
+          pika::execution::experimental::then(std::move(assert_tile_size)) |
+          pika::execution::experimental::start_detached();
```

Comment:
Is there any reason for not using `dlaf::internal::transformDetach`? In case, you could also avoid explicit unwrapping in `assert_tile_size` definition.

Reply:
Only to keep the `launch::sync` semantics, though I'm not sure how important that is.